src/gravitational_waves/gw.cpp

```diff
@@ -0,0 +1,352 @@
+// SPDX-FileCopyrightText: 2024 Lisa Biermann, Margarete Mühlleitner, Rui
+// Santos, João Viana
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+/**
+ * @file gravitational wave class
+ */
+
+#include <BSMPT/gravitational_waves/gw.h>
+
+namespace BSMPT
+{
+
+GravitationalWave::GravitationalWave(BounceSolution &BACalc,
+                                     const int &which_transition_temp)
+{
+  data.transitionTemp = BACalc.CalcTransitionTemp(which_transition_temp);
+  data.PTStrength     = BACalc.GetPTStrength();
+  data.InvTimeScale   = BACalc.GetInvTimeScale();
+  data.vb             = BACalc.GetWallVelocity();
+  data.kappa_sw       = Getkappa_sw(data.PTStrength, data.vb);
+  data.K_sw           = GetK_sw(data.PTStrength, data.vb);
+  data.HR             = GetHR(data.InvTimeScale, data.vb);
+  data.kappa_turb     = CalcEpsTurb(BACalc.GetEpsTurb()) * data.kappa_sw;
+  data.K_turb         = GetK_turb(data.PTStrength, data.kappa_turb);
+  data.gstar          = BACalc.GetGstar();
+  data.Hstar          = BACalc.Hubble_Rate(data.transitionTemp);
+
+  if (data.InvTimeScale < 1)
+  {
+    data.status = "failure";
+    Logger::Write(
+        LoggingLevel::GWDetailed,
+        "beta/H < 1 detected, with beta/H = " +
+            std::to_string(data.InvTimeScale) +
+            ". Transition is assumed to happen, but no GWs calculated.");
+  }
+  else if (data.transitionTemp == -1)
+  {
+    data.status = "failure";
+    Logger::Write(LoggingLevel::GWDetailed,
+                  "Requested transition temperature could not be calculated.");
+  }
+}
+
+GravitationalWave::~GravitationalWave()
+{
+}
+
+/**
+ * @brief CalcEpsTurb calculate epsilon for turbulence contribution
+ * @param epsturb_in is the input value for epsturb. If [0..1] set
+ * to value, for -1 we use the upper bound from
+ * https://arxiv.org/abs/1704.05871
+ * @return value for epsturb
+ */
```

Comment:
Also for this file, shouldnt the documentation go to the header?

Reply:
It also is in the header.

Reply:
Then we can remove it here.

Reply:
Done.